src/palace/manager/integration/patron_auth/saml/provider.py

```diff
@@ -308,66 +313,63 @@ def get_authentication_manager(self):
 
         return authentication_manager
 
-    def remote_patron_lookup(self, subject):
+    def remote_patron_lookup(self, subject: SAMLSubject) -> PatronData:
         """Creates a PatronData object based on Subject object containing SAML Subject and AttributeStatement
 
         :param subject: Subject object containing SAML Subject and AttributeStatement
-        :type subject: api.saml.metadata.Subject
 
         :return: PatronData object containing information about the authenticated SAML subject or
             ProblemDetail object in the case of any errors
-        :rtype: Union[PatronData, ProblemDetail]
+
+        :raises: ProblemDetailException, if there's a problem.
         """
         if not subject:
-            return SAML_INVALID_SUBJECT.detailed("Subject is empty")
-
-        if isinstance(subject, PatronData):
-            return subject
```

Comment:
I couldn't find anything passing in a `PatronData`, so I removed this check.

Reply:
🥇